File: scripts/postinstall.js
#!/usr/bin/env node

/**
 * Post-install script for claude-mpm npm package
 * 
 * Sets up necessary directories and validates dependencies
 */

const fs = require('fs');
const path = require('path');
const os = require('os');
const { execSync } = require('child_process');
const mkdirp = require('mkdirp');

console.log('🚀 Setting up claude-mpm...');

// Create user directories
const userDir = path.join(os.homedir(), '.claude-mpm');
const directories = [
  path.join(userDir, 'agents', 'user-defined'),
  path.join(userDir, 'config'),
  path.join(userDir, 'logs'),
  path.join(userDir, 'templates')
];

directories.forEach(dir => {
  mkdirp.sync(dir);
  console.log(`✅ Created directory: ${dir}`);
});

// Create default config if not exists
const configFile = path.join(userDir, 'config', 'settings.json');
if (!fs.existsSync(configFile)) {
  const defaultConfig = {
    version: '1.0',
    hooks: {
      enabled: true,
      port_range: [8080, 8099]
    },
    logging: {
      level: 'INFO',
      max_size_mb: 100,
      retention_days: 30
    },
    agents: {
      auto_discover: true,
      precedence: ['project', 'user', 'system']
    },
    orchestration: {
      default_mode: 'subprocess',
      enable_todo_hijacking: false
    }
  };
  
  fs.writeFileSync(configFile, JSON.stringify(defaultConfig, null, 2));
  console.log('✅ Created default configuration');
}

// Check for Python
try {
  const pythonVersion = execSync('python3 --version', { encoding: 'utf8' });
  console.log(`✅ Found ${pythonVersion.trim()}`);
} catch (e) {
  console.warn('⚠️  Python 3.8+ is required but not found in PATH');
  console.warn('   Please install Python from https://python.org');
}

// Check for Claude CLI
try {
  execSync('which claude', { encoding: 'utf8' });
  console.log('✅ Found Claude CLI');
} catch (e) {
  console.warn('⚠️  Claude CLI not found in PATH');
  console.warn('   Please install Claude CLI to use claude-mpm');
}

console.log('\n✨ claude-mpm setup complete!');
console.log('\nUsage:');
console.log('  claude-mpm          # Interactive mode');
console.log('  claude-mpm -i "prompt"  # Non-interactive mode');
console.log('  ticket create "Fix bug"  # Create a ticket');
console.log('\nFor more information, visit: https://github.com/claude-mpm/claude-mpm');

File: scripts/post_install.py
#!/usr/bin/env python3
"""
Post-installation setup script for claude-mpm.

This script handles the setup tasks that were previously done in setup.py:
- Create necessary directories
- Build dashboard assets
- Install ticket command wrapper

Usage:
    python scripts/post_install.py
"""

import subprocess
import sys
from pathlib import Path


def create_directories():
    """Create necessary directories for claude-mpm."""
    print("Creating claude-mpm directories...")

    # Create user .claude-mpm directory
    user_dir = Path.home() / ".claude-mpm"
    user_dir.mkdir(exist_ok=True)
    print(f"Created: {user_dir}")

    # Create subdirectories
    subdirs = [
        user_dir / "agents" / "user-defined",
        user_dir / "logs",
        user_dir / "config",
    ]

    for subdir in subdirs:
        subdir.mkdir(parents=True, exist_ok=True)
        print(f"Created: {subdir}")


def build_dashboard_assets():
    """Build dashboard assets using Vite if Node.js is available."""
    print("Building dashboard assets...")

    try:
        build_script = Path(__file__).parent / "build-dashboard.sh"
        if build_script.exists():
            print("Running dashboard build script...")
            result = subprocess.run(
                ["bash", str(build_script)],
                capture_output=True,
                text=True,
                cwd=Path(__file__).parent.parent,
                check=False,
            )
            if result.returncode != 0:
                print(f"Warning: Dashboard build failed: {result.stderr}")
                print(
                    "Dashboard will use individual script files instead of optimized bundles."
                )
            else:
                print("Dashboard assets built successfully")
        else:
            print("Dashboard build script not found, skipping...")
    except Exception as e:
        print(f"Warning: Failed to build dashboard assets: {e}")


def install_ticket_command():
    """Install ticket command wrapper."""
    print("Installing ticket command wrapper...")

    try:
        import site

        # Get the scripts directory
        if hasattr(site, "USER_BASE"):
            scripts_dir = Path(site.USER_BASE) / "bin"
        else:
            scripts_dir = Path(sys.prefix) / "bin"

        scripts_dir.mkdir(exist_ok=True)

        # Create ticket wrapper script
        ticket_script = scripts_dir / "ticket"
        ticket_content = '''#!/usr/bin/env python3
"""Ticket command wrapper for claude-mpm."""
import sys
from claude_mpm.ticket_wrapper import main

if __name__ == "__main__":
    sys.exit(main())
'''
        ticket_script.write_text(ticket_content)
        ticket_script.chmod(0o755)
        print(f"Installed ticket command: {ticket_script}")

    except Exception as e:
        print(f"Warning: Failed to install ticket command: {e}")


def aggregate_agent_dependencies():
    """Run agent dependency aggregation script."""
    print("Aggregating agent dependencies...")

    try:
        script_path = Path(__file__).parent / "aggregate_agent_dependencies.py"
        if script_path.exists():
            print("Running agent dependency aggregation...")
            result = subprocess.run(
                [sys.executable, str(script_path)],
                capture_output=True,
                text=True,
                cwd=Path(__file__).parent.parent,
                check=False,
            )
            if result.returncode != 0:
                print(f"Warning: Agent dependency aggregation failed: {result.stderr}")
            else:
                print("Agent dependencies aggregated successfully")
        else:
            print("Agent dependency aggregation script not found, skipping...")
    except Exception as e:
        print(f"Warning: Failed to aggregate agent dependencies: {e}")


def main():
    """Run all post-installation tasks."""
    print("🚀 Running claude-mpm post-installation setup...")
    print("=" * 50)

    try:
        create_directories()
        print()

        build_dashboard_assets()
        print()

        install_ticket_command()
        print()

        aggregate_agent_dependencies()
        print()

        print("✅ Post-installation setup completed successfully!")

    except Exception as e:
        print(f"❌ Post-installation setup failed: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()
